syntax = "proto3";

package keto.proto;

/**
 * This is an example of a protobuf message.
 */
message SandboxCommandMessage {
    string contract = 1;
    bytes contract_hash = 2;
    string contract_name = 3;
    string contract_namespace = 4;
    bytes model = 5;
    bytes transaction = 6;
    bytes changes = 7;
    int64 available_time = 8;
    int64 elapsed_time = 9;
    float fee_ratio = 10;
}

message HttpRequestEntry {
    string key = 1;
    string value = 2;
}

message HttpRequestMessage {
    string contract = 1;
    string contract_hash = 2;
    string contract_name = 3;
    bytes account_hash = 4;
    repeated bytes roles = 5;
    string method = 6;
    string uri = 7;
    string query = 8;
    repeated HttpRequestEntry request_properties = 9;
    string body = 10;
}

message HttpResponseMessage {
    int64 status = 1;
    int64 content_length = 2;
    string content_type = 3;
    string body = 4;
}